File: 0739-daily-temperatures/0739-daily-temperatures.c
int* dailyTemperatures(int* temperatures, int temperaturesSize, int* returnSize) {
    int *result = (int*)calloc(temperaturesSize, sizeof(int));
    int *stack = (int*)malloc(temperaturesSize * sizeof(int));
    int top = -1;
    *returnSize = temperaturesSize;
    for (int i = 1; i < temperaturesSize; i++) {
        if (temperatures[i-1] < temperatures[i]) {
            result[i-1] = i - (i-1);
            while (top >= 0 && temperatures[stack[top]] < temperatures[i]) {
                result[stack[top]] = i - stack[top];
                top--;
            }
        } else {
            stack[++top] = i - 1;
        }
    }
    free(stack);
    return result;
}

File: 0739-daily-temperatures/0739-daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        Stack<Integer> stack = new Stack<>();
        int result[] = new int[temperatures.length];
        for(int i=0; i<temperatures.length; i++){
            result[i] = 0;
        }
        for(int i=1; i<temperatures.length; i++){
            if(temperatures[i-1] < temperatures[i]){
                result[i-1] = i - (i-1);
                while(!stack.isEmpty() && temperatures[stack.peek()] < temperatures[i]){
                    result[stack.peek()] = i - stack.peek();
                    stack.pop();
                }
            }else{
                stack.push(i-1);
            }
        }
        return result;
    }
}